lib/ecl/tests/ecl_smspec_node.c

```diff
@@ -0,0 +1,127 @@
+  test_assert_int_equal( smspec_node_cmp( region_node , region_node ), 0);
+  test_assert_int_equal( smspec_node_cmp( well_node , well_node ), 0);
+  test_assert_int_equal( smspec_node_cmp( group_node , group_node ), 0);
+  test_assert_int_equal( smspec_node_cmp( block_node , block_node ), 0);
+
+  test_assert_true( smspec_node_cmp( misc_node1 , field_node ) < 0 );
+  test_assert_true( smspec_node_cmp( field_node , region_node ) < 0 );
+  test_assert_true( smspec_node_cmp( region_node , group_node ) < 0 );
+  test_assert_true( smspec_node_cmp( group_node , well_node ) < 0 );
+  test_assert_true( smspec_node_cmp( well_node , segment_node ) < 0 );
+  test_assert_true( smspec_node_cmp( segment_node , block_node ) < 0 );
+  test_assert_true( smspec_node_cmp( block_node , aquifer_node) < 0 );
+  test_assert_true( smspec_node_cmp( aquifer_node , misc_node2 ) < 0 );
+
+  test_assert_true( smspec_node_cmp( field_node, misc_node1) > 0 );
+  test_assert_true( smspec_node_cmp( misc_node2, aquifer_node) > 0 );
+  test_assert_true( smspec_node_cmp( misc_node1, misc_node2) < 0 );
+  test_assert_true( smspec_node_cmp( misc_node2, misc_node1) > 0 );
+
+
+  smspec_node_free( segment_node );
+  smspec_node_free( aquifer_node );
+  smspec_node_free( block_node );
+  smspec_node_free( group_node );
+  smspec_node_free( well_node );
+  smspec_node_free( region_node );
+  smspec_node_free( field_node );
+  smspec_node_free( misc_node1 );
+  smspec_node_free( misc_node2 );
+}
+
+void test_cmp_well() {
+  const int dims[3] = {10,10,10};
+  smspec_node_type * well_node1_1 = smspec_node_alloc( ECL_SMSPEC_WELL_VAR , "W1" , "WOPR" , "UNIT" , ":" , dims , 10 , 0 , 0 );
+  smspec_node_type * well_node1_2 = smspec_node_alloc( ECL_SMSPEC_WELL_VAR , "W2" , "WOPR" , "UNIT" , ":" , dims , 10 , 0 , 0 );
+  smspec_node_type * well_node2_1 = smspec_node_alloc( ECL_SMSPEC_WELL_VAR , "W1" , "WWCT" , "UNIT" , ":" , dims , 10 , 0 , 0 );
+  smspec_node_type * well_node2_2 = smspec_node_alloc( ECL_SMSPEC_WELL_VAR , "W2" , "WWWT" , "UNIT" , ":" , dims , 10 , 0 , 0 );
+  //smspec_node_type * well_node_dummy = smspec_node_alloc( ECL_SMSPEC_WELL_VAR , DUMMY_WELL , "WOPR" , "UNIT" , ":" , dims , 10 , 0 , 0 );
+
+  test_assert_int_equal( smspec_node_cmp( well_node1_1 , well_node1_1 ), 0);
+  test_assert_int_equal( smspec_node_cmp( well_node2_2 , well_node2_2 ), 0);
+
+  test_assert_true( smspec_node_cmp( well_node1_1, well_node1_2) < 0 );
+  test_assert_true( smspec_node_cmp( well_node1_1, well_node2_1) < 0 );
+  test_assert_true( smspec_node_cmp( well_node1_1, well_node2_2) < 0 );
+
+  test_assert_true( smspec_node_cmp( well_node1_2, well_node1_1) > 0 );
+  test_assert_true( smspec_node_cmp( well_node1_2, well_node2_1) < 0 );
+
+  //test_assert_true( smspec_node_cmp( well_node1, well_node_dummy) < 0 );
+
+  smspec_node_free( well_node1_1 );
+  smspec_node_free( well_node2_1 );
+  smspec_node_free( well_node1_2 );
+  smspec_node_free( well_node2_2 );
+  //smspec_node_free( well_node_dummy );
+}
```

Comment:
There are a couple of out-commented lines here, should they be removed? (Perhaps they are removed in a later commit.)